modules/preview-react/side-panel/lib/NewSidePanel/hooks/useSidePanel.ts

```diff
@@ -0,0 +1,52 @@
+import {keyframes} from '@emotion/core';
+import {createHook} from '@workday/canvas-kit-react/common';
+import {SidePanelModel} from './useSidePanelModel';
+
+const createKeyframes = (from: number | string, to: number | string) => {
+  const normalized = {
+    from: typeof from === 'number' ? from + 'px' : from,
+    to: typeof to === 'number' ? to + 'px' : to,
+  };
+
+  return keyframes`
+    from {
+      width: ${normalized.from};
+      min-width: ${normalized.from};
+      max-width: ${normalized.from};
+    } to {
+      width: ${normalized.to};
+      min-width: ${normalized.to};
+      max-width: ${normalized.to};
+    }
+  `;
+};
+
+export const useSidePanel = createHook((model: SidePanelModel, ref) => {
+  const motion = {
+    collapse: createKeyframes(model.state.expandedWidth, model.state.collapsedWidth),
+    expand: createKeyframes(model.state.collapsedWidth, model.state.expandedWidth),
+  };
+
+  const handleAnimationEnd = (event: React.AnimationEvent<HTMLDivElement>) => {
+    console.log('animation end');
+  };
+
+  const handleAnimationStart = (event: React.AnimationEvent<HTMLDivElement>) => {
+    console.log('animation start');
+  };
+
+  const animation =
+    model.state.expanded === 'expanded'
+      ? `${motion.collapse} ease-out 200ms`
+      : `${motion.expand} ease-out 200ms`;
+
+  return {
+    ref,
+    css: {
+      // should Box have an animation prop?
+      animation: `${animation}`,
+    },
+    onAnimationStart: handleAnimationStart,
+    onAnimationEnd: handleAnimationEnd,
```

Comment:
We could also have `width` and `maxWidth` styles live in this hook. But should they be in the CSS prop or should standalone? E.g.
 
 ```ts
     const width =
      localModel.state.expanded === 'expanded'
        ? localModel.state.expandedWidth
        : localModel.state.collapsedWidth;
 
 return {
   ref,
   // provided we add an `animation` prop to `Box`
   animation: '',
   width: width,
   maxWidth: width,
 }
 ```
 
 Or:
 
 ```ts
     const width =
      localModel.state.expanded === 'expanded'
        ? localModel.state.expandedWidth
        : localModel.state.collapsedWidth;
 
 return {
   ref,
   css: {
     animation: '',
     width: width,
     maxWidth: width,
   },
   onAnimationStart: handleAnimationStart,
   onAnimationEnd: handleAnimationEnd,
 }
 ```
 
The first implementation assumes a `Box` API, and merges the styles nicely. But is making that assumption a problem? You couldn't throw in a non-`Box` component without extracting those props into styles from the props and putting them into a style object.

The nice thing about the second solution is that it doesn't assume the element being used, but it will necessarily override the `width` and `maxWidth` Box props.